Source/Core/DolphinQt/Debugger/CodeTraceDialog.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+
+#include "Common/CommonTypes.h"
+
+class CodeWidget;
+class QCheckBox;
+class QLineEdit;
+class QLabel;
+class QComboBox;
+class QListWidget;
+class QListWidgetItem;
+class QSpinBox;
+
+struct CodeTrace
+{
+  u32 address = 0;
+  std::string instruction = "";
+  std::string reg0 = "";
+  std::string reg1 = "";
+  std::string reg2 = "";
+  u32 memory_dest = 0;
+  bool is_store = false;
+  bool is_load = false;
+};
+
+struct TraceOutput
```

Comment:
Hmm this is tricky. TraceOutput is created with multiple variables connected to UI options.  Should ForwardTrace and Backtrace, which uses it to track a register/memory from a recorded CodeTrace, also go in a separate file? In the CodeTrace file?

Reply:
If the TraceOuput is only generated by CodeTrace, it can just be the return value of the CodeTrace member functions (```CodeTrace::ForwardTrace```, ```CodeTrace::BackwardTrace```, ```CodeTrace::Codepath```).